modules/gui/emulator_screen.py

```diff
@@ -124,16 +131,51 @@ def toggle_stepping_mode(self) -> None:
         self._stepping_mode = not self._stepping_mode
         if self._stepping_mode:
 
-            def next_step():
+            def update_back_button():
+                if len(self._step_history) > 1:
+                    self._back_button.config(background="orange", state="normal")
+                else:
+                    self._back_button.config(background="grey", state="disabled")
+
+            def next_step(*args):
                 self._current_step += 1
+                self._step_history.append(context.emulator.get_save_state())
+                update_back_button()
+
+            def previous_step(*args):
+                if len(self._step_history) > 1:
+                    self._current_step -= 1
+                    self._step_history.pop()
+                    save_state = self._step_history[-1]
+                    context.emulator.load_save_state(save_state)
+                    update_back_button()
 
             self._stepping_button = Button(
                 self.window, text="⮞", padx=8, background="red", foreground="white", command=next_step, cursor="hand2"
             )
-            self._stepping_button.place(x=0, y=0)
+            self._stepping_button.place(x=32, y=0)
+
+            self._back_button = Button(
+                self.window, text="⮜", padx=8, foreground="white", command=previous_step, cursor="hand2"
+            )
+            self._back_button.place(x=0, y=0)
+            update_back_button()
+
             self._current_step = 0
+            self._step_history.append(context.emulator.get_save_state())
+
+            bind_reference_space = self.window.bind("<space>", next_step)
+            bind_reference_backspace = self.window.bind("<BackSpace>", previous_step)
+
+            self._stepping_mode_bound_keys.append(("<space>", bind_reference_space))
```

Comment:
May be better to have these user configurable in [profiles/keys.yml](https://github.com/40Cakes/pokebot-gen3/blob/main/profiles/keys.yml)